### PR TITLE
Updates the build to use newer setup, and uses CompileStatic

### LINKED ISSUE #6: Update to Gradle 4.x

### PR DESCRIPTION
**Pull request by ThadHouse**
Closes #6 

CompileStatic helps a good amount with configuration time. And the updated build setup makes it so you can use the `plugins` block with local builds. Instructions were added to the readme to set this up properly as well.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-apply plugin: 'groovy'
-apply plugin: 'maven-publish'
 
 dependencies {
-    compile gradleApi()
-    compile localGroovy()
-    compile 'org.ajoberstar:grgit:1.7.0'
+    compile 'org.ajoberstar:grgit:2.2.0'
     testCompile 'junit:junit:4.12'
-    runtime 'org.ajoberstar:grgit:1.7.0'
```
Comment (333fred) [inline build.gradle]: Why is this no longer needed at runtime?

File: README.md [hunk 2]
```diff
@@ -14,12 +14,27 @@ This plugin is written in Groovy, and therefore needs to have Groovy installed o
...
-```bash 
+```bash
 ./gradlew tasks
 ```
 
 Note that you will not be able to publish to plugins.gradle.org. To get a new version uploaded, contact 333fred.
 
+To publish for local testing, use the following command.
+```bash
+./gradlew publishToMavenLocal -PlocalPublish
```
Comment (333fred) [inline README.md]: Rather than this, is it possible to detect that the publishToMavenLocal task was the one that was called?

Comment (333fred) [reply README.md]: And by this, I mean the `-PlocalPublish`. I know I've done task detection in the past, but I can't remember how exactly.

Comment (ThadHouse) [reply README.md]: I'll take another look. When I was writing this last year by the time I could detect publishToMavenLocal was running it wouldn't let me set those properties. But maybe I missed something.

Comment (ThadHouse) [reply README.md]: Both `gradle.taskGraph.whenReady {taskGraph ->` and `tasks.whenTaskAdded { task ->` happen too late. Its not possible to modify the groupId after the initial creation closure. It doesn't seem to be on any publicly exposed property after the fact.

Comment (333fred) [reply README.md]: Alright then, guess we have to go with the property.

Comment (ThadHouse) [reply README.md]: If I ever find another way to do it, We use it in like 5 repos so I'll definitely update this one too. I'll ask in gradles slack support as well.

**Comment by ThadHouse** [comment]
Note this will conflict with #10, so merge one then let me know so I can fix the other. I still haven't gotten a response from gradle for the -PlocalPublish thing.

**Pull request merged**